src/main/java/org/itsallcode/openfasttrace/report/view/html/HtmlReportSummary.java

```diff
@@ -0,0 +1,56 @@
+package org.itsallcode.openfasttrace.report.view.html;
+
+import java.io.PrintStream;
+
+import org.itsallcode.openfasttrace.report.view.AbstractStreamableViewContainer;
+import org.itsallcode.openfasttrace.report.view.IndentationHelper;
+
+/*-
+ * #%L
+ * OpenFastTrace
+ * %%
+ * Copyright (C) 2016 - 2018 itsallcode.org
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+public class HtmlReportSummary extends AbstractStreamableViewContainer
+{
+    /**
+     * Create a new instance of an {@link HtmlReportSummary}
+     * 
+     * @param stream
+     *            print stream
+     */
+    public HtmlReportSummary(final PrintStream stream)
+    {
+        super(stream);
+    }
+
+    @Override
+    protected void renderBeforeChildren(final int level)
```

Comment:
Idea for refactoring: indentation is a cross cutting concern, maybe extract it into a base class. Maybe ViewContainer return HtmlElement objects with element name, attributes and children that are then rendered?

Reply:
I think in the long run we will go for a template solution anyway. I would rather invest the time into that.